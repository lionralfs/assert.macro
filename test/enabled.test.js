const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  pluginName: 'assert.macro - enabled',
  tests: [
    {
      title: 'Remove import if unused',
      code: `import assert from '../dist/assert.macro';`
    },
    {
      title: 'Keep and rewrite import if used',
      code: `
        import assert from '../dist/assert.macro';
        
        function doSomething(val) {
          assert(val > 0, 'Custom error message');
        }
      `
    },
    {
      title: 'Throws if anything other than the default export is used',
      code: `import assert, { nonExistentExport } from '../dist/assert.macro';`,
      error: function(error) {
        return error instanceof MacroError && /only import assert as default/.test(error.message) && /nonExistentExport/.test(error.message);
      }
    }
  ]
});
