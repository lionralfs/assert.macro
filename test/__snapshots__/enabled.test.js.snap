// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assert.macro - enabled Keep and rewrite import if used: Keep and rewrite import if used 1`] = `
"
import assert from '../dist/assert.macro';

function doSomething(val) {
  assert(val > 0, 'Custom error message');
}

      ↓ ↓ ↓ ↓ ↓ ↓

const assert = require('assert.macro/impl');

function doSomething(val) {
  assert(val > 0, 'Custom error message');
}
"
`;

exports[`assert.macro - enabled Remove import if unused: Remove import if unused 1`] = `
"
import assert from '../dist/assert.macro';

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;

exports[`assert.macro - enabled Throws if anything other than the default export is used: Throws if anything other than the default export is used 1`] = `
"
import assert, { nonExistentExport } from '../dist/assert.macro';

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: You should only import assert as default. You are also importing nonExistentExport.
"
`;

exports[`assert.macro - enabled Throws if not used as a call expression: Throws if not used as a call expression 1`] = `
"
import assert from '../dist/assert.macro';

assert;

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: you must call the macro
"
`;

exports[`assert.macro - enabled Throws used without condition argument: Throws used without condition argument 1`] = `
"
import assert from '../dist/assert.macro';

assert();

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: assert() needs at least 1 argument (the assertion condition).
"
`;
